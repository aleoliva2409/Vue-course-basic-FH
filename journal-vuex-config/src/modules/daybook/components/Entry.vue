<template>
  <div
    class="pointer p-2 entry-container"
    @click="$router.push({ name: 'entry', params: { id: 10 } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">29</span>
      <span class="mx-1 fs-5">Noviembre</span>
      <span class="mx-2 fw-light">2021</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid
      reprehenderit voluptatibus sapiente velit consequatur neque suscipit
      itaque qui sit quia?
    </div>
  </div>
</template>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
  }
  .entry-description {
    font-size: 12px;
  }
}
</style>
