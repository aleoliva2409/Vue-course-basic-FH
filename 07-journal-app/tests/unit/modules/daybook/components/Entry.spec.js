import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';
import { journalState } from '../../../mocks/test-journal-state';

describe('Tests on Entry Component', () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should redirect when is clicked', () => {
    const entryContainer = wrapper.find('.entry-container');
    entryContainer.trigger('click');

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: journalState.entries[0].id,
      },
    });
  });

  test('tests on computed props', () => {
    expect(wrapper.vm.day).toBe(23);
    expect(wrapper.vm.month).toBe('Julio');
    expect(wrapper.vm.year).toBe(2021);
  });
});
