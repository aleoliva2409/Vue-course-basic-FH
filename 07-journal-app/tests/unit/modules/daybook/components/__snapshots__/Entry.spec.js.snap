// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on Entry Component should render correctly 1`] = `
<div class="pointer p-2 entry-container">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">Julio</span><span class="mx-2 fw-light">2021</span></div>
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
